api/src/main/java/ai/djl/nn/AbstractBlock.java

```diff
@@ -92,19 +94,41 @@ public AbstractBlock(byte version) {
     /**
      * Use this to add a child block to this block.
      *
-     * @param name Name of the block, must be unique or otherwise existing children with this name
-     *     are removed, must not be null.
+     * @param name Name of the block, must not be null.
      * @param block The block, must not be null.
      * @param <B> The type of block
      * @return the block given as a parameter - that way the block can be created and reassigned to
      *     a member variable more easily.
      */
-    protected final <B extends Block> B addChildBlock(String name, B block) {
+    public final <B extends Block> B addChildBlock(String name, B block) {
         int childNumber = children.size() + 1;
         children.add(String.format(Locale.ROOT, "%02d%s", childNumber, name), block);
         return block;
     }
 
+    /**
+     * Adds a {@link LambdaBlock} as a child block to this block.
+     *
+     * @param name Name of the block, must not be null.
+     * @param f the function forms the {@link LambdaBlock}
+     * @return the child block
+     */
+    public LambdaBlock addChildBlock(String name, Function<NDList, NDList> f) {
```

Comment:
What do you think about this option for fixing the lambda block display? We can use something like https://github.com/frankfliu/djl/commit/a1b789f2f7379b91ada7bc7e001591e4fe7db20a which adds a name to LambdaBlock. We were thinking about doing it anyway for a few uses (OnnxConverter, TensorBoard, etc.) and I think it will work well here too

Reply:
I think adding a name to LambdaBlock will only work for LambdaBlock and not in general. This is still a good idea as fallback to generic naming. However, the purpose here is to use a contextual name for the child block (the function of this child block in the context of the parent block). So I would store the name with the parent and not with the child, esp. there are usecases where you add a child block multiple times to the neural net (can have multiple parents), and you want to give the child block different names for different purposes.

Reply:
To support https://github.com/frankfliu/djl/commit/a1b789f2f7379b91ada7bc7e001591e4fe7db20a , we can change 

        `sb.append(blockName != null ? blockName : block.getClass().getSimpleName());`

to

		```
if (blockName != null)
			sb.append(blockName);
		else if (block instanceof LambdaBlock)
			sb.append(((LambdaBlock) block).getName());
		else
			sb.append(block.getClass().getSimpleName());
```

Reply:
Yeah, that is what I was thinking we could do. Then, it would take contextualized names like "weight" over Conv2D and use explicit names like "relu" over LambdaBlock.